From 04be5041b17e03eb5b9ea474431369fa6a76b6b8 Mon Sep 17 00:00:00 2001
From: Madhava Reddy Siddareddygari <msiddare@cisco.com>
Date: Wed, 29 Sep 2021 13:53:28 -0700
Subject: [PATCH] Address compilation errors for porting to 5.10 kernel

Signed-off-by: Madhava Reddy Siddareddygari <msiddare@cisco.com>
---
 drivers/net/mdio/mdio-mux.c | 25 +++++++++++++++++--------
 1 file changed, 17 insertions(+), 8 deletions(-)

diff --git a/drivers/net/mdio/mdio-mux.c b/drivers/net/mdio/mdio-mux.c
index 6a1d35402..aa57da1fa 100644
--- a/drivers/net/mdio/mdio-mux.c
+++ b/drivers/net/mdio/mdio-mux.c
@@ -90,16 +90,16 @@ int mdio_mux_init(struct device *dev,
 		  struct mii_bus *mux_bus)
 {
 	struct device_node *parent_bus_node;
-	struct device_node *child_bus_node;
+	struct fwnode_handle *child_bus_node;
 	int r, ret_val;
 	struct mii_bus *parent_bus;
 	struct mdio_mux_parent_bus *pb;
 	struct mdio_mux_child_bus *cb;
 
-	if (!mux_node)
-		return -ENODEV;
-
 	if (!mux_bus) {
+		if (!mux_node)
+			return -ENODEV;
+
 		parent_bus_node = of_parse_phandle(mux_node,
 						   "mdio-parent-bus", 0);
 
@@ -130,16 +130,20 @@ int mdio_mux_init(struct device *dev,
 	pb->mii_bus = parent_bus;
 
 	ret_val = -ENODEV;
-	for_each_available_child_of_node(mux_node, child_bus_node) {
-		int v;
+	device_for_each_child_node(dev, child_bus_node) {
+		u32 v;
+		u32 phy_mask;
 
-		r = of_property_read_u32(child_bus_node, "reg", &v);
+		r = fwnode_property_read_u32(child_bus_node, "reg", &v);
 		if (r) {
 			dev_err(dev,
 				"Error: Failed to find reg for child %pOF\n",
 				child_bus_node);
 			continue;
 		}
+		r = fwnode_property_read_u32(child_bus_node, "phy_mask", &phy_mask);
+		if (r)
+			phy_mask = 0;
 
 		cb = devm_kzalloc(dev, sizeof(*cb), GFP_KERNEL);
 		if (!cb) {
@@ -163,7 +167,12 @@ int mdio_mux_init(struct device *dev,
 		cb->mii_bus->parent = dev;
 		cb->mii_bus->read = mdio_mux_read;
 		cb->mii_bus->write = mdio_mux_write;
-		r = of_mdiobus_register(cb->mii_bus, child_bus_node);
+		cb->mii_bus->phy_mask = phy_mask;
+		if (is_of_node(child_bus_node))
+			r = of_mdiobus_register(cb->mii_bus, to_of_node(child_bus_node));
+		else
+			r = mdiobus_register(cb->mii_bus);
+
 		if (r) {
 			dev_err(dev,
 				"Error: Failed to register MDIO bus for child %pOF\n",
-- 
2.26.2

