Update debian abiname to 6-2

From: Olivier Singla <olivier.singla@broadcom.com>


diff --git a/debian/config/defines b/debian/config/defines
index 9c85fab..3563e79 100644
--- a/debian/config/defines
+++ b/debian/config/defines
@@ -1,5 +1,5 @@
 [abi]
-abiname: 6
+abiname: 6-2
 ignore-changes:
  __cpuhp_*
  bpf_analyzer

diff --git a/debian/control.md5sum b/debian/control.md5sum
index 594492a..3f598c8 100644
--- a/debian/control.md5sum
+++ b/debian/control.md5sum
@@ -25,7 +25,7 @@ eda4623fb10d5ba1cefa8502ebaeb0fd  debian/templates/control.image-dbg.in
 0c86e2579f89e94a6cd144eb98593a5b  debian/templates/control.extra.in
 fe4456d48e3218fb8980c8577d03a7ae  debian/templates/control.config.in
 103ea5fb237e376b511ba638afc41837  debian/templates/control.headers.arch.in
-55e418f2d36dd4465141b93d5901ca04  debian/config/defines
+cd65d8ea2035508cd6f5e4c205678aea  debian/config/defines
 151f2a989308056668d9a162a321c32a  debian/config/i386/defines
 d08ab699e6a25ec0da4cb70efc0f2288  debian/config/ppc64el/defines
 32e264f468daa991fa1bafce9e1eee3c  debian/config/mips64el/defines
